File: couples/python/ordered_prefs.py
from prefs import Prefs

class OrderedPrefs(Prefs):
    def __init__(self, quantity):
        Prefs.__init__(self, quantity)
    
    @staticmethod
    def from_prefs(prefs):
        ordered_prefs = OrderedPrefs(prefs.quantity)
        counter = 0
        for i in range(prefs.quantity):
            counter = 0
            for j in range(prefs.quantity):
                ordered_prefs.set_value(i, prefs.get_value(i, j), counter)
                counter+=1
        return ordered_prefs

File: couples/java/src/zeionara/couples/OrderedPrefs.java
package zeionara.couples;

import java.util.concurrent.ThreadLocalRandom;

public class OrderedPrefs extends Prefs implements DataStore, IntMatrix{
	
	public OrderedPrefs(int quantity){
		super(quantity);	
	}

	public OrderedPrefs(Prefs prefs){
		super(prefs.get_quantity());
		int quantity = prefs.get_quantity();
		int counter = 0;
		for (int i = 0; i < quantity; i++){
			counter = 0;
			for (int j = 0; j < quantity; j++){
				set_value(i, prefs.get_value(i, j), counter);
				counter++;
			}
			
		}	
	}	
}

